docs/website/docs/plus/features/transformations/setup.md

```diff
@@ -0,0 +1,113 @@
+:::
+
+To use this feature, follow these steps:
+
+1. [Configure the `dlt.yml` file](#configure-dltyml-file): define a cache and specify transformations.
+2. [Generate scaffolding](#generate-scaffolding): automatically create transformation templates.
+3. [Modify transformations](#modify-transformations): update the generated Python functions or dbt models.
+4. [Run transformations](#run-transformations): execute them on your data.
+
+## Configure `dlt.yml` file
+
+Before setting up the transformations in the `dlt.yml` file, you need to make sure you have defined the cache.
+
+### Defining the cache
+
+You can find detailed instructions on how to define a cache in the [cache core concept](../../core-concepts/cache.md). Here's an example:
+
+```yaml
+caches:
+  github_events_cache:
+    inputs:
+      - dataset: github_events_dataset
+        tables:
+          items: items
+    outputs:
+      - dataset: github_events_dataset
+        tables:
+          items: items
+          items_aggregated: items_aggregated
+```
+
+:::caution
+Please, make sure that the input dataset for the cache is located in a filesystem-based destination ([Iceberg](../../ecosystem/iceberg.md), [Delta](../../ecosystem/delta.md), or [Cloud storage and filesystem](../../../dlt-ecosystem/destinations/filesystem.md))
+:::
+
+### Defining transformations
+
+Specify transformations in `dlt.yml` with the following parameters:
+
+* unique identifier for the transformation.
+* engine – choose between:
+  * `arrow` for Python-based transformations
+  * `dbt` for dbt-based transformations
+* cache – the cache that the transformation will run on.
+
+For example,
+
+```yaml
+transformations:
+  github_events_transformations:
+    engine: dbt
+    cache: github_events_cache
+```
+
+## Generate scaffolding
+
+To create transformation scaffolding based on your dlt pipeline:
+
+1. Run the dlt pipeline at least once; this ensures dlt has the dataset schemas.
+2. Execute the following CLI command:
+
+```sh
+dlt transformation <transformation-name> render-t-layer
+```
+
+This will generate transformation files inside the `./transformations` folder. Depending on the engine:
+
+* For Python transformations: a Python script with transformation functions ([learn more](./python-transformations.md))
+* For dbt transformations: dbt models ([learn more](./dbt-transformations.md))
+
+Each generated transformation includes models for managing incremental loading states via `dlt_load_id`.
+
+## Modify transformations
+
+Now you can update the generated transformations and create new ones to reflect the desired behavior. We recommend keeping the incremental approach as in the generated models.
+
+## Run transformations
+
+dlt+ offers comprehensive CLI support for executing transformations. You can find the full list of available commands in the [command line interface](../../reference.md).
+
+To run all transformations, use the following command:
```

Comment:
the following command will only run the first found transformation, not all of them. we should probably not show the dot notation...